var/spack/repos/builtin/packages/mesa-demos/package.py

```diff
@@ -3,26 +3,61 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+import sys
+
 from spack import *
 
 
 class MesaDemos(AutotoolsPackage):
     """This package provides some demo applications for testing Mesa."""
 
     homepage = "https://www.mesa3d.org"
-    url      = "https://github.com/freedesktop/mesa-demos/archive/mesa-demos-8.3.0.tar.gz"
+    url      = "https://github.com/freedesktop/mesa-demos/archive/mesa-demos-8.3.0.tar.gz"  # Should this be from free desktop gitlab?
 
     version('8.3.0', sha256='9bc1b37f4fc7bfc3f818f2d3851ffde28e8167ef11dca87f4781e9ef6206901f')
     version('8.2.0', sha256='5a9f71b815d968d0c3b77edfcc3782d0211f8520b00da9e554ccfed80c8889f6')
     version('8.1.0', sha256='cc5826105355830208c90047fc38c5b09fa3ab0045366e7e859104935b00b76d')
 
+    variant('osmesa', default=True, description="Enable the OSMesa frontend.")
+
+    is_linux = sys.platform.startswith('linux')
+    variant('glx', default=is_linux, description="Enable the GLX frontend.")
+
     depends_on('autoconf',  type='build')
     depends_on('automake',  type='build')
     depends_on('libtool',   type='build')
     depends_on('m4',        type='build')
     depends_on('pkgconfig', type='build')
-    depends_on('mesa')
+
+    depends_on('gl')
+    depends_on('glx', when='+glx')
```

Comment:
I believe there is also a dependency on `libx11` and `libxext` when `+glx` is enabled